File: demo/demo/src/main/java/com/example/demo/service/UserService.java
package com.example.demo.service;


import com.example.demo.model.Role;
import com.example.demo.model.User;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.util.List;

public interface UserService {

    User saveUser(User user);
    Role saveRole(Role role);
    void addRoleToUser(String username, String roleName);
    User getUser(String username);
    User updateUser(String firstName, String lastName, String email, String age, String password);
    User getById(long id);
    List<User> getUsers();
    boolean deleteUser(long id);
    void refreshToken(HttpServletRequest request, HttpServletResponse response) throws IOException;
}

File: frontend/src/services/user.service.js
import axios from 'axios';
import authHeader from './auth-header';
import authService from './auth.service';

const USER_URL = 'http://localhost:8080/api/users/';
const GETUSER_URL = 'http://localhost:8080/api/user/';
const UPDATEUSER_URL = 'http://localhost:8080/api/update?';
const DELETEUSER_URL = 'http://localhost:8080/api/user/';


class UserService {
  
  getUsers() {
    return axios.get(USER_URL, {headers:authHeader()});
  } 
  getUser() {
    return axios.get(GETUSER_URL + authService.getUsername(), {headers:authHeader()});
  }

  updateUser(firstName, lastName, email, age, password) {
    return axios.post(UPDATEUSER_URL+"firstName="+firstName+
                    "&lastName="+lastName+"&email="+email+"&age="+age+
                    "&password="+password
    , "",{headers:authHeader()});
  }

  deleteUser(id) {
    return axios.delete(DELETEUSER_URL + id, {headers:authHeader()});
  }





}
export default new UserService();